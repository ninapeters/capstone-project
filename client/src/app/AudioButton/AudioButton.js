import styled from 'styled-components/macro'
import PropTypes from 'prop-types'
import { PlayIcon, PauseIcon } from '../Icons'

AudioButton.propTypes = {
  onClick: PropTypes.func,
  isSongPlaying: PropTypes.bool.isRequired,
  isSmall: PropTypes.bool,
}

export default function AudioButton({ onClick, isSongPlaying, isSmall }) {
  return (
    <Button onClick={onClick} data-testid="audio-button" isSmall={isSmall}>
      {isSongPlaying ? <PauseIcon title="pause" /> : <PlayIcon title="play" />}
    </Button>
  )
}

const Button = styled.button`
  background: ${(props) =>
    props.isSmall ? 'transparent' : 'var(--white-transparent-min)'};
  border: none;
  border-radius: 50%;
  box-shadow: ${(props) => (props.isSmall ? 'none' : 'var(--shadow-light)')};
  fill: var(--primary-regular);
  height: ${(props) => (props.isSmall ? '48px' : '60px')};
  padding: 14px;
  width: ${(props) => (props.isSmall ? '48px' : '60px')};
  &:focus {
    outline: 0;
  }
`
