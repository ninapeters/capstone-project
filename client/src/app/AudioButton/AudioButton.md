Audio Button: Play

```jsx
<AudioButton onClick={() => {}} isSongPlaying={false} />
```

Audio Button: Pause

```jsx
<AudioButton onClick={() => {}} isSongPlaying={true} />
```
