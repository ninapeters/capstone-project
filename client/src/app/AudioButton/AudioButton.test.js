import { render } from '@testing-library/react'
import user from '@testing-library/user-event'
import AudioButton from './AudioButton'

describe('AudioButton', () => {
  it('renders correctly', () => {
    const { container } = render(
      <AudioButton onClick={() => {}} isSongPlaying={false} />
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('shows play button per default', () => {
    const { isSongPlaying, getByTitle } = render(
      <AudioButton onClick={() => {}} isSongPlaying={false} />
    )
    expect(isSongPlaying).toBeFalsy()
    expect(getByTitle('play')).toBeInTheDocument()
  })

  it('calls handleClick with the correct id', () => {
    const onClickMock = jest.fn()
    const { getByTestId } = render(
      <AudioButton onClick={() => onClickMock('a')} isSongPlaying={false} />
    )
    const button = getByTestId('audio-button')
    user.click(button)

    expect(onClickMock).toHaveBeenCalledWith('a')
  })

  it('changes the button appearance by rerender', () => {
    const { getByTitle, queryByTitle, rerender } = render(
      <AudioButton onClick={() => {}} isSongPlaying={false} />
    )
    expect(getByTitle('play')).toBeInTheDocument()

    rerender(<AudioButton onClick={() => {}} isSongPlaying={true} />)
    expect(queryByTitle('play')).not.toBeInTheDocument()
    expect(getByTitle('pause')).toBeInTheDocument()
  })
})
