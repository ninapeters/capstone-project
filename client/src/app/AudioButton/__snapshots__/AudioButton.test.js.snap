// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AudioButton renders correctly 1`] = `
.c0 {
  background: var(--white-transparent-min);
  border: none;
  border-radius: 50%;
  box-shadow: var(--shadow-light);
  fill: var(--primary-regular);
  height: 60px;
  padding: 14px;
  width: 60px;
}

.c0:focus {
  outline: 0;
}

<button
  class="c0"
  data-testid="audio-button"
>
  <svg
    title="play"
  >
    play.svg
  </svg>
</button>
`;
